config/envs/rhpam-authoring-ha.yaml

```diff
@@ -138,17 +250,172 @@ others:
           labels:
             app: "[[.ApplicationName]]"
             application: "[[.ApplicationName]]"
-          name: "[[.ApplicationName]]-amq"
+            service: "[[.ApplicationName]]-amq-scaledown-controller"
+          name: "[[.ApplicationName]]-amq-scaledown-controller"
+          annotations:
+            deployment.kubernetes.io/revision: "1"
+            template.alpha.openshift.io/wait-for-ready: "true"
         spec:
           replicas: 1
           selector:
-            deploymentConfig: "[[.ApplicationName]]-amq"
+            deploymentConfig: "[[.ApplicationName]]-amq-scaledown-controller"
           strategy:
             rollingParams:
               maxSurge: 0
             type: Rolling
           template:
             metadata:
+              labels:
+                app: "[[.ApplicationName]]"
+                application: "[[.ApplicationName]]"
+                deploymentConfig: "[[.ApplicationName]]-amq-scaledown-controller"
+              name: "[[.ApplicationName]]-amq-scaledown-controller"
+            spec:
+              serviceAccountName: "[[.ApplicationName]]-amq-scaledown-controller-sa"
+              containers:
+                - args:
+                    - --localOnly
+                    - --v
+                    - "3"
+                  image: "amq-broker-72-scaledown-controller-openshift"
+                  imagePullPolicy: IfNotPresent
+                  name: "[[.ApplicationName]]-amq"
+              terminationGracePeriodSeconds: 30
+          triggers:
+            - imageChangeParams:
+                automatic: true
+                containerNames:
+                  - "[[.ApplicationName]]-amq"
+                from:
+                  kind: ImageStreamTag
+                  namespace: "openshift"
+                  name: "amq-broker-72-scaledown-controller-openshift:0.4"
```

Comment:
Why there is a specified version 0.4 here when template is using 1.0? https://github.com/jboss-container-images/rhpam-7-openshift-image/blob/7.3.x/templates/rhpam73-authoring-ha.yaml#L288

The 1.0 version from template is mapped to 0.4 in AMQ image stream.

Reply:
@sutaakar where in that template is the imagestream mapping to 0.4 created?

Reply:
@tchughesiv You can find it in https://raw.githubusercontent.com/jboss-container-images/jboss-amq-7-broker-openshift-image/amq-broker-72/amq-broker-7-scaledown-controller-image-streams.yaml

Reply:
@sutaakar ok, but are we saying this imagestream will be available in next 3.11 release? 4.0?

Reply:
I am not sure when it becomes available. Just pointing how is the image stream defined in our templates and how it is mapped in AMQ image stream file.

Reply:
i get it... but the issue we have here is, what if using the operator in a cluster that doesn't have those AMQ imagestreams? the DC will attempt to pull amq-broker-72-scaledown-controller-openshift:1.0 from our registry. That image doesn't exist so the env will never deploy. Additionally, the operator has the ability to create missing imagestreams for the user... but if it tries to create one to a non-existent image, that's no use. 

I can certainly change it to 1.0 but not sure it makes much sense to.

Reply:
hmm, I see. In that case I would leave it as it is now, can be adjusted later.

Reply:
There is obviously a difference between an image _stream_ version and the _image_ version/tag that it references. In this case, as you noted, the AMQ project's image streams file defines a 1.0 stream that points to a 0.4 image:
https://github.com/jboss-container-images/jboss-amq-7-broker-openshift-image/blob/amq-broker-72/amq-broker-7-scaledown-controller-image-streams.yaml#L26-L29
Personally, I wish they were the same. Specifically, I wish they were both 0.4, so that they can both become 1.0 at the same time. However, it's not my project...
Maybe the best thing to do is see how the stream is versioned in the OCP catalog? Or ask the maintainers of that project?

Reply:
i see your point too though... let's just let @errantepiphany make the call :)

Reply:
https://github.com/jboss-container-images/jboss-amq-7-broker-openshift-image/graphs/contributors

Reply:
@sutaakar tks for calling this out... i've made the switch to `1.0` as you suggested.